### PR TITLE
Autocrop

### PR DESCRIPTION
**Pull request by PeterChaplin**
Needs testing (and checklist)

### CODE REVIEW

**Comment by PeterChaplin** [comment]
Testing
----
- Local make-install, restart server
- Test with https://github.com/qcode-software/mla-static/pull/25 and https://github.com/qcode-software/mla-www/pull/1111
- View homepage with un-cropped images
- Open a product image in a new tab,
- Modify uri to autocrop uri
- Check that resulting image is auto-cropped
- Flush memoization, reload homepage
- Check that autocropped image is used.

**Comment by PeterChaplin** [comment]
Tcl
----
- [x] Does each proc have a #| comment to describe what it does ?
- [x] Are there comments throughout the code ?
- [x] Is each proc unit testable ?
- [x] Are the proc names well chosen ?
- [x] Are variable names well chosen ?
- N/A Security: Is user input sanitised ?
- [N/A Security: Is substitution protected from injection attacks ?
- [x] List what testing has been done.

File: tcl/image_file.tcl [hunk 1]
```diff
@@ -219,41 +270,81 @@ proc qc::image_cache_create {args} {
...
+    {image_redirect_handler "qc::image_redirect_handler"}
+    {allowed_max_width 2560}
+    {allowed_max_height 2560}
+} {
     #| URL handler to serve images that can not be served by fastpath.
     # Create image cache for canonical URL if it doesn't already exist.
-    # If canonical URL was requested return file to client and register URL to be servered by fastpath for future requests.
-    # Otherwise default redirect handler will redirect client to correct image dimesions or the canonical URL.
+    # If canonical URL was requested return file to client,
+    # and register URL to be servered by fastpath for future requests.
```
Comment (nicky-johnstone) [inline tcl/image_file.tcl]: Looks like there were some typos in master. Could you fix them while we're here?

`servered` -> `served`

File: tcl/image_file.tcl [hunk 2]
```diff
@@ -219,41 +270,81 @@ proc qc::image_cache_create {args} {
...
+    # Otherwise default redirect handler will redirect client,
+    # to correct image dimesions or the canonical URL.
```
Comment (nicky-johnstone) [inline tcl/image_file.tcl]: `dimesions` -> `dimensions`

**Pull request merged**